azerothcore/azerothcore-wotlk
# chore(Map) Add open world LOS settings

## Changes Proposed:

This PR proposes changes to:

## Issues Addressed:

- Closes https://github.com/andeswow/bugtracker/issues/3

## Tests Performed:

This PR has been:

## How to Test the Changes:


1. Enter with a character in the open world
2. By default the configuration variable will be disabled, so you can traverse mountains, trees and possibly some other structure.
3. By enabling the config variable and reloading the server configuration `.reload config`, you will notice that it is no longer possible to shoot, heal or perform any type of action, behind a mountain, tree or other structure in the open world.
4. Since this configuration is not blizzlike, by default it would be disabled, and it will depend on each community whether it is enabled or not. Many communities have it, and many communities ask for it. So it would be good to be able to add it.

**For default (disabled)**

https://github.com/azerothcore/azerothcore-wotlk/assets/2810187/1c810dad-ed2f-4633-8788-4697c303578f

**Enabled**

https://github.com/azerothcore/azerothcore-wotlk/assets/2810187/d7386c28-61cf-4377-803d-edd614c5c37c

## Known Issues and TODO List:



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
